block-producer/src/batch_builder/mod.rs

```diff
@@ -120,7 +121,7 @@ where
         let batch = Arc::new(TransactionBatch::new(txs)?);
         self.ready_batches.write().await.push(batch);
 
-        info!(COMPONENT, "batch built with {num_txs} txs");
```

Comment:
I strongly prefer the usage of structured logging instead of formatted strings. In this case we should keep the `COMPONENT` key and add `num_txs` as a key.

IMO this structured logging is best because it allows the log data to be collected to a database (say, elastic search), which parses the logs and make it queriable.

Here this would look like this:

```diff
info!(num_txs=txs, COMPONENT, "batch built");
```

Reply:
I prefer structured logging as well. Two questions:

1. How far can we push it? For example, here would be nice to capture IDs of transactions that went into the batch. Could we provide a vector of transaction IDs as another field?
2. How does the structured output look like when it is printed out to the console (assuming we still use pretty-printer for that)?

Reply:
> How does the structured output look like when it is printed out to the console (assuming we still use pretty-printer for that)?

The rendering the spans/events is completely customizable via a [`Subscriber`](https://docs.rs/tracing/latest/tracing/trait.Subscriber.html) API. Currently the code is using one from [tracing_subscriber](https://github.com/0xPolygonMiden/miden-node/blob/main/utils/src/logging.rs#L6-L19), which itself has a few default [formatters](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/fmt/index.html#formatters). The current is set to [`pretty`](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/fmt/format/struct.Pretty.html) with `compact`, which looks like this:

![image](https://github.com/0xPolygonMiden/miden-node/assets/310139/195d41eb-8f33-430f-8e7a-070977bcf3f0)

Reply:
@hackaugusto 
> I strongly prefer the usage of structured logging instead of formatted strings. In this case we should keep the `COMPONENT` key and add `num_txs` as a key.

Thank you for noticing! I just missed `num_txs`, will extract it from the string.
I will also rollback `COMPONENT`, since I removed it due to confusion (I though, it's used as `target`, which is disabled by us on subscriber, but now I see, this is a separate field).

@bobbinth
> 1. How far can we push it? For example, here would be nice to capture IDs of transactions that went into the batch. Could we provide a vector of transaction IDs as another field?

Yes, we can log any data types which implement `Debug` or `Display` traits.